packages/admin-ui/src/lib/core/src/providers/alerts/alerts.service.ts

```diff
@@ -91,8 +102,26 @@ export class AlertsService {
     }
 
     configureAlert<T>(config: AlertConfig<T>) {
-        this.alertsMap.set(config.id, new Alert(config));
-        this.configUpdated.next();
+        this.hasSufficientPermissions(config.requiredPermissions)
+            .pipe(
+                filter(canReadPendingSearchIndexUpdates => canReadPendingSearchIndexUpdates),
```

Comment:
If I understand this code correctly: 

If the `hasSufficientPermissions()` observable emits `false`, this filter will block the emitted value from ever reaching the `subscribe()` function, and the subscription will stay open and not complete. 

One way to avoid this would be to remove the `filter()` clause and inside the subscriber you can inspect the value and only set the map if the value is `true`.

Reply:
I thought the user data might refresh periodically. This would eventually add the alert once permissions change. However the data is only loaded once - so you are obviously right.

I moved the check into the subscription as you suggested. The observable will now always close.
I need to skip the first `userStatus` value as permissions have not been fetched yet. I therefore added a filter for the `userStatus` stream:
```ts
filter(({ userStatus }) => userStatus.isLoggedIn),
```